hw1/02-palindrome.js

```diff
@@ -1,3 +1,25 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
 
-elem.addEventListener('input', handleInput);
+elem.addEventListener("input", handleInput);
+
+function handleInput() {
+  const value = parseFloat(elem.value);
+  const resultDiv = document.querySelector("#result");
+
+  if (value >= 0) {
+    const numString = value.toString();
+    const reverseString = numString.split("").reverse().join("");
+
+    if (numString == reverseString) {
+      resultDiv.textContent = "Yes. This is a palindrome!";
+      document.getElementById("result").style.color = "green";
+    } else {
+      resultDiv.textContent = "No. Try again.";
+      document.getElementById("result").style.color = "red";
+    }
+  } else {
+    const resultDiv = document.querySelector("#result");
+    resultDiv.textContent = "Please enter a positive number.";
```

Comment:
checking for negative or invalid number is missing.

Reply:
You're right, no explicit error checking for a negative is present, however, a positive integer is the only thing accepted and the user will be notified with any other input.